File: ruby/hamming/hamming.rb
class Hamming
  def self.compute s1, s2
    dist = 0
    if s1.length != s2.length
      raise ArgumentError
    end
    s1.chars.each_with_index {|v, i| dist+=1 if v != s2[i]}
    dist
  end
end

File: javascript/hamming/hamming.js
var Hamming  = function(){
}

Hamming.prototype.compute = function(strd1, strd2){
  var distance = 0;
  if(strd1.length === strd2.length && strd2.length !== 0){
    for(var i = 0; i < strd1.length; i++){
      if (strd1.charAt(i) !== strd2.charAt(i)){ distance++; }
    }
  }else{
    throw 'DNA strands must be of equal length.'
  }
  return distance;
}

module.exports = Hamming;
